### PR TITLE
Submit completed builds

### LINKED ISSUE #6: Project completed submission form `/grants/{database_id}/submit`
When a builder's project is already validated, they can submit the result. (Github)

### PR DESCRIPTION
**Pull request by Pabl0cks**
## Description

Tried to follow the same logic than current status changes.

- Created a button with a modal in /my-grants for the user to submit the build of their active grant. Tried different designs, not so happy with it, could ask Andrea for a future iteration.
- Added notifications when submitting, and logic to save `submittedAt` timestamp.
- Added link to grant card in /admin page, not very fancy link, only underlined text to the right of grant title.
- Wanted to use a stackoverflow simple URL validation but did not find any that I was happy with, so is chatgpt crafted. Take a special look at it at the review 👀

Feel free to push any changes that you see necessary, or let me know if you prefer me to do them ♥🙌

Closes #6

### CODE REVIEW

**Comment by technophile-04** [comment]
Hey tysm Pablo !! This is great!

I think we missed the signing of the message part, If we we don't add that part anyone can impersonate anyone's address and submit on behalf of them.

So we need to sign EIP712 message on frontend and also verify it on backend. Similar to the flow how we have it for `/apply`page.

You could reference this for frontend :
https://github.com/BuidlGuidl/grants.buidlguidl.com/blob/6fea69365a2f9ac8c604fbac48109c98173866ad/packages/nextjs/app/apply/_component/Form.tsx#L43

And this for backend verification:
https://github.com/BuidlGuidl/grants.buidlguidl.com/blob/6fea69365a2f9ac8c604fbac48109c98173866ad/packages/nextjs/app/api/grants/new/route.ts#L30

Please let us know if you need any help!! Thanks ! 🙌

**Comment by technophile-04** [comment]
Update few things:

- Moved modal to its separate component
   - We could in future PR's refactor it more on UI side if needed / use dasiyUI modal but... looks great for now
- Added EIP717 signature auth
- Added more authorization (lol just to be sure someone does not bad req with Postman) : 
  - Only can be submitted if it was approved
  - Only grant owner can submit it

File: packages/nextjs/app/api/grants/submit/route.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+type ReqBody = {
+  grantId?: string;
+  signer?: string;
+  link?: string;
+  signature?: `0x${string}`;
+};
+
+// TODO: Maybe we could fetch build from firebase and check if its really present,
+// also check that build was actually submitted by the same builder
```
Comment (technophile-04) [inline packages/nextjs/app/api/grants/submit/route.ts]: cc @carletex, should we add this check too?

File: packages/nextjs/app/api/grants/submit/route.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+type ReqBody = {
+  grantId?: string;
+  signer?: string;
+  link?: string;
+  signature?: `0x${string}`;
+};
+
+// TODO: Maybe we could fetch submmited build on BG app from firebase and check if its really present,
+// also check that build was actually submitted by the same builder
```
Comment (technophile-04) [inline packages/nextjs/app/api/grants/submit/route.ts:15]: cc @carletex, should we add this check too or lol its too much ?

Comment (carletex) [reply packages/nextjs/app/api/grants/submit/route.ts:15]: haha, I think it's not worth it.

**Review by carletex — APPROVED** [review]
Great stuff @technophile-04 & @Pabl0cks !!

**Pull request merged**